backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java

```diff
@@ -2,20 +2,45 @@
 
 import com.htecgroup.skynest.model.response.ErrorMessage;
 import com.htecgroup.skynest.util.DateTimeUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.context.request.WebRequest;
+import java.util.ArrayList;
+import java.util.List;
 
 @ControllerAdvice
+@Slf4j
 public class UserExceptionHandler {
 
   @ExceptionHandler(value = {UserException.class})
   public ResponseEntity<ErrorMessage> handleUserException(UserException ex, WebRequest webRequest) {
     ErrorMessage errorMessage =
         new ErrorMessage(
             ex.getMessage(), ex.getStatus().value(), DateTimeUtil.currentTimeFormatted());
-
+    log.error(ex.toString(), ex);
     return new ResponseEntity<>(errorMessage, ex.getStatus());
   }
+
+  @ExceptionHandler(value = {MethodArgumentNotValidException.class})
+  public ResponseEntity<ErrorMessage> handleValidationException(
+      MethodArgumentNotValidException ex) {
+    List<String> errors = new ArrayList<>();
+    ex.getBindingResult()
+        .getAllErrors()
+        .forEach(
+            error ->
+                errors.add(
+                    String.format(
+                        "%s %s", ((FieldError) error).getField(), error.getDefaultMessage())));
```

Comment:
Instead of forEach, take a look at **_map_** and _**collect**_ options and use them as combination.